packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -589,6 +589,11 @@ See documentation: ${createDocumentationLink({
         persistHierarchicalRootCount: this.future.persistHierarchicalRootCount,
       });
 
+    if (this.compositionID) {
+      mainHelper.searchForFacetValues =
+        mainHelper.searchForCompositionFacetValues.bind(mainHelper);
+    }
```

Comment:
I went with this approach because it allows things to be completely transparent for `connectRefinementList`, but I'm open to suggestions